pandas/core/ops/array_ops.py

```diff
@@ -157,8 +159,14 @@ def _na_arithmetic_op(left, right, op, is_cmp: bool = False):
     """
     import pandas.core.computation.expressions as expressions
 
+    if isinstance(right, str):
+        # can never use numexpr
+        func = op
```

Comment:
Basically with a string argument, numexpr will fail with a "wrong" error message. Alternatively, `_can_use_numexpr` in expressions.py could also be updated to check for this and avoid using the numexpr path (currently that only checks object with dtypes, not for scalars)

Reply:
lets make an effort to keep numexpr-specific lgoic in _can_use_numexpre/expressions

Reply:
@jbrockmendel would you be OK with leaving the check here as is, short term? I have a next PR that moves this check inside a `can_use_numexpr` function inside `expressions.py` (https://github.com/pandas-dev/pandas/pull/41122), so that will clean this up. 
But I would like to merge this PR before https://github.com/pandas-dev/pandas/pull/41122 since this one is adding a lot of test coverage for with/without numexpr.

Reply:
yeah ok for now, but let's for sure move later